linker.go

```diff
@@ -190,17 +242,15 @@ func fixupAndValidate(insns asm.Instructions) error {
 		fixupProbeReadKernel(ins)
 	}
 
-	if err := fixupKfuncs(insns); err != nil {
-		return fmt.Errorf("fixing up kfuncs: %w", err)
-	}
-
 	return nil
 }
 
 // fixupKfuncs loops over all instructions in search for kfunc calls.
 // If at least one is found, the current kernels BTF is loaded to set Instruction.Constant
 // to the running kernels btf id of the btf.Func in the instructions Metadata for all kfunc call instructions.
-func fixupKfuncs(insns asm.Instructions) error {
+func fixupKfuncs(insns asm.Instructions) (*handlesMap, error) {
+	fdArray := newHandlesMap()
```

Comment:
I'd move the declaration of `fdArray` to before the for loop once we actually find kfuncs. In error cases you can return `nil` for `handles`.

Reply:
:ballot_box_with_check: